kafka-rest/src/main/java/io/confluent/kafkarest/controllers/ProduceControllerImpl.java

```diff
@@ -53,7 +56,13 @@
             timestamp.toEpochMilli(),
             key.map(ByteString::toByteArray).orElse(null),
             value.map(ByteString::toByteArray).orElse(null),
-            /* headers= */ emptyList()),
+            headers.entries().stream()
+                .map(
+                    header ->
+                        new RecordHeader(
+                            header.getKey(),
+                            header.getValue().map(ByteString::toByteArray).orElse(null)))
```

Comment:
noted elsewhere too - don't understand prevalent use of ByteString over byte array.

Reply:
Ack.